app/helpers/admin_api_helper.rb

```diff
@@ -19,8 +19,8 @@ def not_found_error(_exception)
     render json: { error: "Not found" }, status: :not_found
   end
 
-  def not_valid_error(_exception)
-    render json: { error: "Not valid" }, status: :bad_request
+  def not_valid_error(exception)
+    render json: { error: exception.message }, status: :bad_request
```

Comment:
I'm not sure I follow this commit. What was missing from the exception messages previously to make them un-useful? ie why does using the application id make the difference between them being useful or not?

Reply:
Previously the error message was a generic "Not valid", after this change the returned error message will be (e.g. [from test](https://github.com/alphagov/signon/pull/1768/commits/b620dd4d900c1a914946de5feb4282b24107a63d#diff-0e0ce7463121fd2e27cdc242c28a84fa04413d272034d54c78f8019000062b5eR71)) "Validation failed: Redirect URI must be an absolute URI." which clients can use to more easily debug why a request failed.

Reply:
Sure, I just wondered why we were previously setting a response of "Not valid" rather than returning `exception.message`

Reply:
Yeah I'm not sure why I didn't do this previously 🤔

Reply:
Ah OK that's fine! I thought there was a clever reason 😂